### PR TITLE
DT-2582 Disallow removing of immutable adminTypes on role

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: backend/controllers/roleAdminTypeAmendment.js [hunk 1]
```diff
@@ -15,31 +20,34 @@ const roleAdminTypeAmendmentFactory = (getRoleDetailsApi, changeRoleAdminTypeApi
...
       roleUrl,
-      currentRoleAdminType: adminTypeMap,
+      adminTypeValues,
+      currentFilter: roleAdminType.map((e) => e.adminTypeCode),
       errors: req.flash('changeRoleErrors'),
     })
   }
 
+  const maintainImmutableAdminTypes = (originalAdminType, amendedAdminType) => {
+    const allImmutableAdminTypes = adminTypeValues.filter((item) => item.immutable === true).map((e) => e.value)
```
Comment (brightonsbox) [inline backend/controllers/roleAdminTypeAmendment.js:33]: Can the === true be dropped here?

Comment (karenmillermoj) [reply backend/controllers/roleAdminTypeAmendment.js:33]: I think I tried that but it failed...will try again now all code is back

**Review by brightonsbox — DISMISSED** [review]
LGTM

File: backend/controllers/roleAdminTypeAmendment.js (hunk 1 above)
Comment (brightonsbox) [reply backend/controllers/roleAdminTypeAmendment.js:33]: It might not realise it's a boolean, no worries if it doesn't work

Comment (karenmillermoj) [reply backend/controllers/roleAdminTypeAmendment.js]: Have checked and the boolean js check can be updated neatly - have pushed that fix back to this branch, but will need an additional review

**Review by brightonsbox — APPROVED** [review]
LGTM

**Pull request merged**